manifests/pulp.pp

```diff
@@ -46,15 +59,39 @@
     repo_auth              => true,
     puppet_wsgi_processes  => 1,
     enable_katello         => true,
+    https_cert             => $https_cert,
+    https_key              => $https_key,
+    ca_cert                => $ca_cert,
     subscribe              => Class['certs', 'certs::qpid_client'],
     worker_timeout         => $pulp_worker_timeout,
   }
 
   contain ::pulp
 
-  foreman::config::passenger::fragment { 'pulp':
-    content     => file('katello/pulp-apache.conf'),
-    ssl_content => file('katello/pulp-apache-ssl.conf'),
+  if $manage_httpd {
```

Comment:
This is still not really great. The third use case is katello devel which has a third vhost. I think a better solution would be to directly use apache vhost concat fragments and just pass in the title of the vhost.